### ISSUE TITLE
Spring MVC gets empty locale

### ISSUE THREAD

**Issue by smtalk** [nginx/unit#479]
Hello,

I don't see any reports of https://mkyong.com/spring-mvc/spring-mvc-internationalization-example/ not working with unit because of blank ${pageContext.response.locale}. I guess unit might have a problem similar to https://bugs.eclipse.org/bugs/show_bug.cgi?id=351039 ?

I haven't debugged it deeper yet, if you need more details - just let me know.

Thank you.

**Comment by smtalk** [comment]
@mar0x Hm.. the client says his application, which works fine in tomcat, is still having problems. Is Accept-Language header passed to the application? Did https://mkyong.com/spring-mvc/spring-mvc-internationalization-example/ work fine for you?

Please let me know if you need more details collected (including debug mode, code samples etc.). He said he tried multiple approaches and none has worked (including https://www.baeldung.com/spring-boot-internationalization and the mentioned one).

Thank you.

**Comment by mar0x** [comment]
@smtalk Need more detailed description of "having problems". There is no special processing of `Accept-Language` header and assume it is passed to application "as is".

I've built and run spring-mvc example and I can see 'Current Locale' is not null. I've got same response as running this app in Jetty.

Does client use latest Unit from git or official mercurial repo? The fix is not in release version (1.20.0) yet.

If client uses latest Unit and "still having problems", it would be useful to know what responses he expect for particular requests (using `curl` is more useful than browser navigations).

Here is the sample script to download, build and run example in Unit:

```shell
#/bin/bash

# Stop on error exit status.
set -e

# download example
curl -O -A 'Mozilla' 'https://www.mkyong.com/wp-content/uploads/2010/08/SpringMVC-Internationalization-Example.zip'

# unzip example
unzip SpringMVC-Internationalization-Example.zip

# patch pom
patch -p0 -b << EOF
--- SpringMVC/pom.xml	2010-08-03 09:51:58.000000000 +0400
+++ SpringMVC/pom.xml	2020-10-14 13:10:48.000000000 +0300
@@ -41,9 +41,21 @@
 		<artifactId>standard</artifactId>
 		<version>1.1.2</version>
 	</dependency>
+
+<dependency>
+    <groupId>javax.servlet</groupId>
+    <artifactId>javax.servlet-api</artifactId>
+    <version>3.1.0</version>
+    <scope>provided</scope>
+</dependency>
  
   </dependencies>
+
   <build>
     <finalName>SpringMVC</finalName>
   </build>
+  <properties>
+      <maven.compiler.source>1.7</maven.compiler.source>
+      <maven.compiler.target>1.7</maven.compiler.target>
+  </properties>
 </project>
EOF

# package app
cd SpringMVC
mvn package
cd ../

# get unit
curl -o unit-master.zip 'https://codeload.github.com/nginx/unit/zip/master'
unzip unit-master
cd unit-master/

# configure
./configure --debug --modules=build --state=build --tmp=build
./configure java

# make binaries
make -j4 -s

# start unit
./build/unitd

cat << EOF > ../SpringMVC.conf
{
  "applications": {
    "SpringMVC": {
      "type": "java",
      "webapp": "../SpringMVC/target/SpringMVC.war",
    },
  },

  "listeners": {
    "*:8400": {
      "pass": "applications/SpringMVC"
    },
  }
}
EOF

# apply runtime configuration
curl -X PUT -d @../SpringMVC.conf --unix-socket control.unit.sock http://localhost/config

# make test requests
curl -v 'http://127.0.0.1:8400/welcome.htm'
curl -v 'http://127.0.0.1:8400/welcome.htm?language=zh_CN'

# stop unit
kill `cat unit.pid`
```
